src/core/ticker/TickerListener.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * Internal class for handling the priority sorting of ticker handlers.
+ *
+ * @private
+ * @class
+ * @memberof PIXI.ticker
+ */
+export default class TickerListener
+{
+    /**
+     * Constructor
+     *
+     * @param {Function} fn - The listener function to be added for one update
+     * @param {Function} [context=null] - The listener context
+     * @param {number} [priority=0] - The priority for emitting
+     * @param {boolean} [once=false] - If the handler should fire once
+     */
+    constructor(fn, context = null, priority = 0, once = false)
+    {
+        /**
+         * The handler function to execute.
+         * @member {Function}
+         */
+        this.fn = fn;
+
+        /**
+         * The calling to execute.
+         * @member {Function}
+         */
+        this.context = context;
+
+        /**
+         * The current priority.
+         * @member {number}
+         */
+        this.priority = priority;
+
+        /**
+         * If this should only execute once.
+         * @member {boolean}
+         */
+        this.once = once;
+
+        /**
+         * The next item in chain.
+         * @member {TickerListener}
+         */
+        this.next = null;
+
+        /**
+         * The previous item in chain.
+         * @member {TickerListener}
+         */
+        this.previous = null;
+
+        /**
+         * Marked for cleanup.
+         * @member {boolean}
+         */
+        this.destroyed = false;
+    }
+
+    /**
+     * Simple compare function to figure out if a function and context match.
+     *
+     * @param {Function} fn - The listener function to be added for one update
+     * @param {Function} context - The listener context
+     * @return {boolean} `true` if the listener match the arguments
+     */
+    match(fn, context)
+    {
+        context = context || null;
+
+        return this.fn === fn && this.context === context;
+    }
+
+    /**
+     * Emit by calling the current function.
+     * @param {number} deltaTime - time since the last emit.
+     */
+    emit(deltaTime)
+    {
+        if (this.destroyed)
+        {
+            return;
+        }
```

Comment:
Although most of the time, I am all for defensive programming, in my opinion, this should just be left to cause an exception for the case this is checking for. This is an internal implementation, consumers wouldn't accidentally do this, and if the internal implementation does, something is horribly wrong.